web/src/engine/build.sh

```diff
@@ -0,0 +1,201 @@
+# making it a third param.  This is currently unclear, but if so, we'd do
+# $DEBUG/$INTERMEDIATE and $RELEASE/$INTERMEDIATE via a third argument.
+output_path ( ) {
+  if [ $# -lt 1 ]; then
+    echo "Insufficient argument count!"
+    exit 1
+  elif [ $# -eq 1 ]; then
+    # Used by clean:<target> actions
+    echo "$BUILD_BASE/$1"
+  else
+    echo "$BUILD_BASE/$1/$2"
+  fi
+}
+
+SOURCE="src"
+
+# Ensures that we rely first upon the local npm-based install of Typescript.
+# (Facilitates automated setup for build agents.)
+PATH="../../../node_modules/.bin:$PATH"
+
+compiler="npm run tsc --"
+compilecmd="$compiler"
+
+builder_check_color "$@"
+
+# TODO: for predictive-text, we only need :headless, perhaps we should be splitting modules?
+# TODO: remove :tools once kmlmc is a dependency for test:module
+
+DOC_WEB_PRODUCT="${BUILDER_TERM_START}:web${BUILDER_TERM_END} build product"
+DOC_TEST_WEB="${BUILDER_TERM_START}test:web${BUILDER_TERM_END}"
+DOC_BUILD_EMBED_WEB="${BUILDER_TERM_START}build:embed${BUILDER_TERM_END} and ${BUILDER_TERM_START}build:web${BUILDER_TERM_END}"
+DOC_TEST_SYMBOL="actions - ${BUILDER_TERM_START}test${BUILDER_TERM_END}"
+
+builder_describe "Builds engine modules for Keyman Engine for Web (KMW)." \
+  "@../../../common/web/keyman-version build:engine build:embed build:web build:ui" \
+  "@../../../common/web/input-processor build:engine build:embed build:web" \
+  "clean" \
+  "configure" \
+  "build" \
+  ":device-detect     Subset used for device-detection " \
+  ":element-wrappers  Subset used to integrate with website elements" \
+  ":main              Builds all common code used by KMW's app/-level targets"
+
+# Possible TODO?
+# "upload-symbols   Uploads build product to Sentry for error report symbolification.  Only defined for $DOC_BUILD_EMBED_WEB" \
+
+builder_describe_outputs \
+  configure                  ../node_modules \
+  configure:device-detect    ../node_modules \
+  configure:element-wrappers ../node_modules \
+  configure:main             ../node_modules \
+  build:device-detect        $(output_path $DEVICEDETECT $OUTPUT_DIR)/index.js \
+  build:element-wrappers     $(output_path $ELEMENTWRAPPERS $OUTPUT_DIR)/index.js \
+  build:main                 $(output_path $MAIN $OUTPUT_DIR)/keymanweb.js
+
+builder_parse "$@"
+
+#### Build utility methods + definitions ####
+
+# Compiles all build products corresponding to the specified target.
+#
+# ### Parameters
+#
+# * 1: `product`    the product's source path under src/
+#
+# ### Example
+#
+# ```bash
+#   compile engine/main
+# ```
+compile ( ) {
+  if [ $# -lt 1 ]; then
+    fail "Scripting error: insufficient argument count!"
+  fi
+
+  local COMPILE_TARGET=$1
+  local COMPILED_OUTPUT_PATH=$(output_path $COMPILE_TARGET $OUTPUT_DIR)
+
+  BUNDLE_CONFIG=src/$COMPILE_TARGET/tsconfig.bundled.json
+
+  $compilecmd -b src/$COMPILE_TARGET -v
+
+  # Handle any 'bundled' compilations, too!
+  if [ -f "$BUNDLE_CONFIG" ]; then
```

Comment:
A deficiency has been noted here address by https://github.com/keymanapp/keyman/pull/7831/commits/9c569b76230be8f3b68c92b617a68f76b37ad904 in the next PR, at least at present.

tl;dr:  I made a bad assumption about the working directory for the if-check due to running in a subdirectory.  The following `$compilecmd` isn't affected b/c it involves an `npm` script call.